src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -1698,6 +1698,11 @@ bool QgsAdvancedDigitizingDockWidget::canvasKeyPressEventFilter( QKeyEvent *e )
 
 void QgsAdvancedDigitizingDockWidget::clear()
 {
+  if ( mCurrentTool )
+  {
+    mCurrentTool->deleteLater();
+  }
+
```

Comment:
I'm afraid this is not enough to handle the right clicking case.
We should be capturing the right click in `QgsAdvancedDigitizingCirclesIntersectionTool` and consume it so it never ends at the actual mapTool.
Consider the following example:
- User digitizes a line, has already added two points
- selects 2 circle intersection
- if he right clicks, his line is finalized, while he should actually end up just like having pressed `Esc`: continue digitizing

Reply:
@uclaros , right I wasn't quite sure what would be the ideal option here. You think it'd be better for it not to act as a finalization action when this tool is active?

Reply:
For 2 circle intersection, probably not. I'll have the abstract tool able to either take or pass event, and for this intersection tool I'll take (and cancel the cogo operation as suggested here).

Reply:
@uclaros , done.